### PR TITLE
Support for PVR Providers and VOD Media as recordings

### PR DESCRIPTION
**Pull request by phunkyfish**
v20.1.0
- Fixed: Change reported backend name
- Added: Support pvr providers in Kodi 20
- Added: Support VOD Media as recordings

### CODE REVIEW

File: src/PVRIptvData.cpp [hunk 1]
```diff
@@ -87,17 +93,19 @@ PVR_ERROR PVRIptvData::GetCapabilities(kodi::addon::PVRCapabilities& capabilitie
   capabilities.SetSupportsTV(true);
   capabilities.SetSupportsRadio(true);
   capabilities.SetSupportsChannelGroups(true);
+  capabilities.SetSupportsProviders(true);
   capabilities.SetSupportsRecordings(false);
   capabilities.SetSupportsRecordingsRename(false);
   capabilities.SetSupportsRecordingsLifetimeChange(false);
   capabilities.SetSupportsDescrambleInfo(false);
+  capabilities.SetSupportsRecordings(true);
```
Comment (AlwinEsch) [inline src/PVRIptvData.cpp:100]: On line 97 already the `capabilities.SetSupportsRecordings(false);` present, better there change to true.

Comment (AlwinEsch) [reply src/PVRIptvData.cpp:100]: also seems then here a `  capabilities.SetSupportsRecordingsDelete(false);` missing.

File: README.md [hunk 2]
```diff
@@ -163,6 +172,21 @@ Settings related to Channel Logos.
...
+### Media
+Media entries can be used to access the Video-On-Demand (VOD) library provided by IPTV services. The IPTV media library can be organized either as a plain list of channels or the hierarchy tree made of folders and PVR recordings.
+
+An M3U entry can denote that it's media by having:
+* an M3U property of `EXT-X-PLAYLIST-TYPE` set to `VOD`,
+* the M3U attributes of `media`, `media-dir` or `media-size` are set
+
+More detail on these can be found in [Supported M3U and XMLTV elements](#supported-m3u-and-xmltv-elements).
+
+* **Show media as recordings**: If enabled, all IPTV media entries can be showm as PVR recordings. Otherwise, they appear as regular PVR channels.
```
Comment (basilgello) [inline README.md]: typo: s/showm/shown/

File: README.md [hunk 3]
```diff
@@ -320,6 +344,17 @@ Note that both these files are provided as examples and are overwritten each tim
 
 The format is quite simple, containing a number of channel group/bouquet names.
 
+### Provider Name Mappings (Channels)
+
+Config files are located in the `userdata/addon_data/pvr.vuplus/providers` folder.
```
Comment (basilgello) [inline README.md]: typo:s/vuplus/iptvsimple/

File: README.md [hunk 4]
```diff
@@ -455,6 +503,14 @@ http://path-to-stream/live/channel-z.ts
   - `catchup-source`: Can contain the full catchup URL (complete with format specifiers) if in `default` mode. Or if the mode is `append` just the query string with format specifiers which will be appended to the channel URL. If omitted the `Query format string` from settings will be appended.
   - `catchup-days`: The number of days in the past catchup is available for.
   - `catchup-correction`: A correction to the time used for catchup stream URL generation. Useful for catchup streams which are geo mis-matched to the wrong time.
+  - `provider`: The name that idenitifes this provider and will be seen in the provider manager. Also that name that can be used for any provider mappings if required. No other provider are valid if this tag is omitted.
```
Comment (basilgello) [inline README.md]: What is meant by "no other provider are valid if this tag is omitted"? Maybe "No other provider **tags** are valid if this tag is omitted" ?

File: README.md [hunk 5]
```diff
@@ -455,6 +503,14 @@ http://path-to-stream/live/channel-z.ts
...
+  - `provider-type`: A type which must match one of addon, satellite, cable, aerial, iptv or unknown.
```
Comment (basilgello) [inline README.md]: Isn't it better to highlight valid types with list of code syntax: A type which must match one of `addon`, `satellite`, `cable`, `aerial`, `iptv` or `unknown`

File: README.md [hunk 6]
```diff
@@ -455,6 +503,14 @@ http://path-to-stream/live/channel-z.ts
...
+  - `provider-logo`: A path to the location where the icon for this provider is available.
+  - `provider-countries`: The country for this provider. Should be passed using an ISO-3166 country code. eave empty or omit for unspecified.
```
Comment (basilgello) [inline README.md]: I'd make "ISO-3166" a link pointing to table of country codes or just use example in braces, just like `provider-languages` below. Also typo: s/eave/leave/

**Comment by basilgello** [comment]
Works for me with Divan.TV (Ukraine)

**Comment by phunkyfish** [comment]
> Works for me with Divan.TV (Ukraine)

Great, thanks for testing!

**Pull request closed without merging**